### PR TITLE
feat(ts) Convert placeholder component to typescript

### PR DESCRIPTION
**Pull request by markstory**

### CODE REVIEW

File: src/sentry/static/sentry/app/components/placeholder.tsx [hunk 1]
```diff
@@ -4,33 +4,45 @@ import PropTypes from 'prop-types';
 
 import space from 'app/styles/space';
 
-const Placeholder = styled(props => {
+const defaultProps = {
+  shape: 'rect' as 'rect' | 'circle',
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/placeholder.tsx]: why do we need `as 'rect'` here?

*edit* I see it's trying to cast it as `'rect'|'circle'` (and that `defaultProps` is not a type)

File: src/sentry/static/sentry/app/components/placeholder.tsx [hunk 2]
```diff
@@ -4,33 +4,45 @@ import PropTypes from 'prop-types';
...
+})<Props>`
   display: flex;
   flex-direction: column;
   flex-shrink: 0;
   justify-content: center;
 
   background-color: ${p => p.theme.placeholderBackground};
-  width: ${p => p.width};
-  height: ${p => p.height};
+  width: ${p => p.width || '100%'};
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/placeholder.tsx]: Isn't this default?

**Review by billyvg — APPROVED** [review]

File: src/sentry/static/sentry/app/components/placeholder.tsx (hunk 1 above)
Comment (markstory) [reply src/sentry/static/sentry/app/components/placeholder.tsx]: Yeah without the `as` statement typescript would infer the type as string, when I want it to be just rect or circle.

File: src/sentry/static/sentry/app/components/placeholder.tsx (hunk 2 above)
Comment (markstory) [reply src/sentry/static/sentry/app/components/placeholder.tsx]: I'll remove this. I put this in before I figured out how to get defaultProps to typecheck.

**Pull request merged**